api/api-common/src/main/kotlin/io/qalipsis/api/meters/steps/impl/CounterMeterStepSpecificationImpl.kt

```diff
@@ -0,0 +1,70 @@
+/*
+ * QALIPSIS
+ * Copyright (C) 2025 AERIS IT Solutions GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package io.qalipsis.api.meters.steps.impl
+
+import io.micronaut.core.annotation.Introspected
+import io.qalipsis.api.context.StepContext
+import io.qalipsis.api.meters.Counter
+import io.qalipsis.api.meters.steps.failure.impl.ComparableValueFailureSpecification
+import io.qalipsis.api.meters.steps.CounterMeterStepSpecification
+import io.qalipsis.api.meters.steps.failure.CounterFailureConditionSpec
+import io.qalipsis.api.meters.steps.failure.impl.CounterFailureConditionSpecImpl
+import io.qalipsis.api.steps.AbstractStepSpecification
+import io.qalipsis.api.steps.StepSpecification
+
+/**
+ * Implementation of [CounterMeterStepSpecification].
+ *
+ * @author Francisca Eze
+ */
+@Introspected
+data class CounterMeterStepSpecificationImpl<INPUT>(
```

Comment:
One tip. move the CounterMeterStepSpecification in the same file to have all the hierarchy in the same place. Name the file according to the interface name, not the implementation.

Do the same of all the steps (and only the steps).